File: test_python_project/src/core_module/actions/add_task_to_list.py
from utils.base_action import BaseBusinessAction
from core_module.actions.get_tasks_list import GetTasksList

class AddTaskToList(BaseBusinessAction):
    def execute(
        self,
        task: dict,
    ) -> list[int]:
        list = GetTasksList().execute(user_id=1)
        self._internal_preconditions_check()
        return list

    def _internal_preconditions_check(self):
        list_to_check = GetTasksList().execute(user_id=1)
        pass

    def _internal_empty(self):
        pass

File: test_typescript_project/src/core_module/actions/add_task_to_list.ts
import { makeTaskId, type Task } from "@/core_module/domain/task";
import { InMemoryTasksRepo } from "@/core_module/repo/in_memory_tasks_repo";
import { BaseBusinessAction } from "@/utils/base_action";

export type AddTaskInput = Readonly<{ title: string }>;
export type AddTaskOutput = Readonly<{ task: Task }>;

export class AddTaskToList extends BaseBusinessAction<
  AddTaskInput,
  AddTaskOutput
> {
  constructor(private readonly repo = new InMemoryTasksRepo()) {
    super();
  }

  async execute(input: AddTaskInput): Promise<AddTaskOutput> {
    const task: Task = {
      id: makeTaskId(`task_${Date.now()}`),
      title: input.title,
      done: false,
      createdAt: new Date(),
    };

    await this.repo.add(task);
    return { task };
  }
}
